### PR TITLE
chore: refactor _convert_to_standard_langcode

### PR DESCRIPTION
**Pull request by Coniferish**
This PR is one in a series of PRs for refactoring and fixing the `languages` parameter so it can address incorrect input by users. #2293 

This PR adds a dictionary for helping map fully spelled out languages to tesseract language codes

### CODE REVIEW

File: unstructured/partition/utils/constants.py [hunk 1]
```diff
@@ -39,6 +39,156 @@ class PartitionStrategy:
...
+    "chinese - simplified": "chi_sim",
+    "chinese": "chi_sim",
+    "chinese - traditional": "chi_tra",
+    "cherokee": "chr",
+    "corsican": "cos",
+    "welsh": "cym",
+    "danish": "dan",
+    "danish - fraktur": "dan_frak",
+    "german": "deu",
+    "german - fraktur (contrib)": "deu_frak",  # "contrib" not removed because it would repeat key
```
Comment (shreyanid) [inline unstructured/partition/utils/constants.py:74]: why would removing contrib repeat a key? can't "german" and "german - fraktur" be structured in the same way as "danish" and "danish - fraktur"?

**Comment by shreyanid** [comment]
I'm not actually seeing a difference in output between main and this branch - I'm running this code in a temp `shreya_test.py`, switching out the languages to be "english", "spanish" and "angry" (just as a random non language word)
```
from unstructured.partition.auto import partition

elements = partition(filename="example-docs/language-docs/eng_spa_mult.doc", languages=["angry"])
print([el.metadata.languages for el in elements])
```
and am getting
```
(unst10) shreyanid@Shreyas-MacBook-Pro-2 unstructured % python shreya_test.py
[['eng'], ['eng'], ['eng'], ['eng'], ['eng']]
(unst10) shreyanid@Shreyas-MacBook-Pro-2 unstructured % python shreya_test.py
[['spa'], ['spa'], ['spa'], ['spa'], ['spa']]
(unst10) shreyanid@Shreyas-MacBook-Pro-2 unstructured % python shreya_test.py
[['ang'], ['ang'], ['ang'], ['ang'], ['ang']]
```

This behavior is the same on both main and this branch. Is this full language name already supported? Could you provide test instructions if there is a different behavior to be testing for this PR?

File: unstructured/partition/utils/constants.py (hunk 1 above)
Comment (Coniferish) [reply unstructured/partition/utils/constants.py:74]: Because there's another key with "german - fraktur" that has a different value. See line 90.

**Comment by Coniferish** [comment]
> I'm not actually seeing a difference in output between main and this branch - I'm running this code in a temp `shreya_test.py`, switching out the languages to be "english", "spanish" and "angry" (just as a random non language word)
> 
> ```
> from unstructured.partition.auto import partition
> 
> elements = partition(filename="example-docs/language-docs/eng_spa_mult.doc", languages=["angry"])
> print([el.metadata.languages for el in elements])
> ```
> 
> and am getting
> 
> ```
> (unst10) shreyanid@Shreyas-MacBook-Pro-2 unstructured % python shreya_test.py
> [['eng'], ['eng'], ['eng'], ['eng'], ['eng']]
> (unst10) shreyanid@Shreyas-MacBook-Pro-2 unstructured % python shreya_test.py
> [['spa'], ['spa'], ['spa'], ['spa'], ['spa']]
> (unst10) shreyanid@Shreyas-MacBook-Pro-2 unstructured % python shreya_test.py
> [['ang'], ['ang'], ['ang'], ['ang'], ['ang']]
> ```
> 
> This behavior is the same on both main and this branch. Is this full language name already supported? Could you provide test instructions if there is a different behavior to be testing for this PR?

This is a refactor with a minor update for mapping languages that are found in the dictionary that's added. It's not intended to change the functionality in a significant way. I did add one test showing it correctly maps "spanish" to "spa"

**Comment by shreyanid** [comment]
Ok, thanks for the clarification! Yeah, definitely seeing how the index slice of the input string results in "angry" being reduced to "ang" and interpreted as a language heh, perhaps that can be a future quick fix.

**Review by shreyanid — APPROVED** [review]
LGTM! Not required, but if we want to allow full language names as inputs now based on this change, we can reflect that in the input format / docstring for the `languages` parameter

**Comment by Coniferish** [comment]
Great.
I don't think we do. This is just to catch common mistakes by users. In a later PR, I'll be adding more warning logs

**Pull request merged**